src/main/java/io/cdap/plugin/gcp/dataplex/common/util/DataplexUtil.java

```diff
@@ -589,4 +593,44 @@ private static Type dataplexLogicalFieldType(Schema schema) {
 
     return null;
   }
+
+  /**
+   * Add partition info to dataplex entity.
+   *
+   * @param entity dataplex entity
+   * @param credentials Google Credentials
+   * @param bucketName
+   * @param tableName
+   * @param project
+   * @throws IOException
+   */
+  public static void addPartitionInfo(Entity entity,
+                                      GoogleCredentials credentials,
+                                      String bucketName, String tableName, String project) throws IOException {
+    Storage storage = GCPUtils.getStorage(project, credentials);
+    String delimiter = "/";
+    String partitionPrefix = DataplexConstants.STORAGE_BUCKET_PARTITION_KEY + "=";
+    Page<Blob> blobs =
+      storage.list(
+        bucketName,
+        Storage.BlobListOption.prefix(tableName + delimiter + partitionPrefix),
+        Storage.BlobListOption.currentDirectory());
+    String lastPartition = null;
+    for (Blob blob : blobs.iterateAll()) {
```

Comment:
Not too familiar with iterables in Java. Is there a better way (i.e., in O(1)) to get to the last element of the iterator (looks like it's a linked list not an array)?

Reply:
I don't think there's a way to reverse iterate over Pages